unstructured/embed/embedder/open_ai.py

```diff
@@ -0,0 +1,155 @@
+import json
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional
+
+from dataclasses_json import DataClassJsonMixin
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.ingest.error import EmbedderConnectionError
+from unstructured.ingest.interfaces import (
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    IngestDocSessionHandleMixin,
+)
+from unstructured.ingest.logger import logger
+from unstructured.partition.json import partition_json
+from unstructured.utils import requires_dependencies
+
+
+@requires_dependencies(["langchain"])  # add extras="langchain" when it's added to the makefile
+@dataclass
+class OpenAISessionHandle(BaseSessionHandle):
+    from langchain.embeddings.openai import OpenAIEmbeddings
+
+    service: OpenAIEmbeddings
+
+
+@EmbedderConnectionError.wrap
+@requires_dependencies(["langchain"])  # add extras="langchain" when it's added to the makefile
+def create_openai_object(api_key, model_name):
+    """
+    Creates an OpenAI object to embed elements.
+    Args:
+        api_key: API Key, generated for the user
+        model_name: Model name for OpenAI
+
+    Returns:
+        OpenAI Embeddings object
+    """
+    from langchain.embeddings.openai import OpenAIEmbeddings
+
+    embedder = OpenAIEmbeddings(
```

Comment:
usually don't want to comment on names at this stage but... I'd propose using `encoder` instead of `embedder`. `encoder` is more commonly used.

Reply:
I see we have `encoder` elsewhere... welp

Reply:
> I see we have `encoder` elsewhere... welp

I hadn't thought about that while naming it, but I agree that it might be confused with `ElementEncoder` if we name it as encoder

This object manages the requests to OpenAI API - I wanted to name it as `embedder` so users with less ML experience can also make sense of it (due to it being used as a trendy word in wider circles)

Reply:
how about we just call this `embeddings`, per the class name and that seems to be the name they give to the instance in [their documentation](https://api.python.langchain.com/en/latest/embeddings/langchain.embeddings.openai.OpenAIEmbeddings.html).

Reply:
Changed the namings with https://github.com/Unstructured-IO/unstructured/pull/1368/commits/499795ae9364593a4954a30e0e568598f3d15f33, please let me know if we'd like further changes on the names